### DISCUSSION TITLE
Starting Orange from pycharm

### DISCUSSION THREAD

**Discussion by leilyzafari** [biolab/orange3#5267] [Q&A] [answered]
Hi,

I am developing a new add-on for Orange3. 
At the moment I am debugging the code by opening the Orange Gui, run my widget and look at the debug messages on the console or the "show log" window.
Is there a way to run Orange from inside pycharm so that we can pause on a commands and look into more detail?

**Comment by ajdapretnar** [comment, marked as answer]
Running the entire Orange in PyCharm works by setting the Run/Debug Configuration.
I call it `main`. The settings are:
Script path: `/Users/user/orange/orange3/Orange/canvas/__main__.py`
Parameters: `--clear-widget-settings`

For running an individual widget, use:
```
if __name__ == "__main__":
    data = Table("iris")
    WidgetPreview(OWDataTable).run(set_dataset=data)
```

**Comment by janezd** [comment]
`--clear-widget-settings` is optional. I actually have two configurations, one with and one without it.

All properly written widgets contains the code that @ajdapretnar quoted, so you can run them in debuger.

**Comment by irgolic** [comment]
PyCharm lets you commit run/debug configurations, we should do that.
